pkg/service/providerindex/legacy.go

```diff
@@ -21,17 +29,19 @@ type LegacyClaimsFinder interface {
 type LegacyClaimsStore struct {
 	contentToClaims ContentToClaimsMapper
 	claimsStore     types.ContentClaimsStore
+	claimsUrl       string
 }
 
 // ContentToClaimsMapper maps content hashes to claim cids
 type ContentToClaimsMapper interface {
 	GetClaims(ctx context.Context, contentHash multihash.Multihash) (claimsCids []cid.Cid, err error)
 }
 
-func NewLegacyClaimsStore(contentToClaimsMapper ContentToClaimsMapper, claimStore types.ContentClaimsStore) LegacyClaimsStore {
+func NewLegacyClaimsStore(contentToClaimsMapper ContentToClaimsMapper, claimStore types.ContentClaimsStore, claimsUrl string) LegacyClaimsStore {
```

Comment:
yeah, that's a good idea. I thought about an alternative when I started working on this: instead of returning plain `ProviderResult`s, define a new struct that allows attaching the actual claim to the provider record. Then, in the indexing service, check if the result has a claim attached and use that instead of getting it from the claims service.

This alternative requires modifying the code of the indexing service to make it aware of claims attachments, but avoids having using the cache to communicate the legacy claims service and the "regular" claims service, which feels... wrong?

Anyway, since the idea is to remove this code eventually, your idea of passing the cache to the legacy claims service may make more sense.

Reply:
Can be separate PR.